vendor/github.com/docker/libnetwork/endpoint.go

```diff
@@ -528,7 +528,8 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 	}()
 
 	if sb.needDefaultGW() && sb.getEndpointInGWNetwork() == nil {
-		return sb.setupDefaultGW()
+		err = sb.setupDefaultGW()
+		return err
```

Comment:
Oh, I see the defer now. That's kind of ugly. If a defer is going to inspect the value of `err`, the function should use a "named return":

```go
func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) (err error) {
```

Then returning an error will automatically assign it to `err`.